edx_arch_experiments/datadog_monitoring/docs/decisions/0001-monitoring-by-code-owner.rst

```diff
@@ -0,0 +1,54 @@
+Monitoring by Code Owner
+************************
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+It is difficult for different teams to have team-based on-calls rotations, alerting and monitoring for various parts of the edx-platform (specifically LMS).
+
+* The original decision to add code_owner custom span tags (custom attributes) was documented in `edx-platform in 0001-monitoring-by-code-owner`_.
+* The decision to move the code for reuse across IDAs was documented in `edx-django-utils in 0001-monitoring-by-code-owner.rst`_.
+* The decision for how to implement code owner details for celery tasks was documented in `0003-code-owner-for-celery-tasks_, and was limited by New Relic's instrumentation.
+* The decision to break up the ``code_owner`` custom span tag (custom attribute) into ``code_owner_squad`` and ``code_owner_theme`` tags was documented in `0004-code-owner-theme-and-squad`_.
+
+Some changes or clarifications since this time:
+
+* It turns out that this functionality is only really useful for the edx-platform LMS. Our other services (IDAs) are small enough to keep to a single owner, or to solve monitoring issues in other ways.
+* It is likely that the ``code_owner`` code is only really needed by 2U.
+* 2U wants to drop owner themes from its code_owner custom span tag.
+* 2U has switched to Datadog, which has slightly different capabilities from New Relic.
+
+  * Note that Datadog has custom span tags, where New Relic has custom attributes to refer to its tagging capabilities.
+
+.. _edx-platform in 0001-monitoring-by-code-owner: https://github.com/openedx/edx-platform/blob/f29e418264f374099930a5b1f5b8345c569892e9/lms/djangoapps/monitoring/docs/decisions/0001-monitoring-by-code-owner.rst
+.. _edx-django-utils in 0001-monitoring-by-code-owner.rst: https://github.com/openedx/edx-django-utils/blob/a1a1ec95d7c1d4767deb578748153c99c9562a04/edx_django_utils/monitoring/docs/decisions/0001-monitoring-by-code-owner.rst
+.. _0003-code-owner-for-celery-tasks: https://github.com/openedx/edx-django-utils/blob/a1a1ec95d7c1d4767deb578748153c99c9562a04/edx_django_utils/monitoring/docs/decisions/0003-code-owner-for-celery-tasks.rst
+.. _0004-code-owner-theme-and-squad: https://github.com/openedx/edx-django-utils/blob/a1a1ec95d7c1d4767deb578748153c99c9562a04/edx_django_utils/monitoring/docs/decisions/0004-code-owner-theme-and-squad.rst
+
+Decision
+========
+
+2U has moved its code owner monitoring implementation to the datadog_monitoring plugin.
+
+* The owner theme name has been dropped from the ``code_owner`` custom span tag value in this new implementation.
+* The ``code_owner_theme`` span tag has been dropped altogether from this new implementation.
+* The now deprecated ``code_owner_squad`` span tag, which is redundant with the updated ``code_owner`` tag, will continue to be supported for backward compatibility.
```

Comment:
Perhaps just temporarily until we update any docs and monitors/dashboards?

```suggestion
* The now deprecated ``code_owner_squad`` span tag, which is redundant with the updated ``code_owner`` tag, will continue to be supported temporarily for backward compatibility.
```

Reply:
I went back-and-forth, but I'm going to leave this as-is. It may be permanent if we never choose to update everything, since this is in terraform templates, etc. And if we choose to go from deprecated to removal, the lack of the word "temporarily" doesn't change the fact that this is documented as deprecated.